pipework

```diff
@@ -249,61 +249,59 @@ MTU=$(ip link show $IFNAME | awk '{print $5}')
+  ip link add link "$IFNAME" dev "$GUEST_IFNAME" mtu "$MTU" type macvlan mode bridge
+  ip link set "$IFNAME" up
 }
 
-ip link set $GUEST_IFNAME netns $NSPID
-ip netns exec $NSPID ip link set $GUEST_IFNAME name $CONTAINER_IFNAME
-[ "$MACADDR" ] && ip netns exec $NSPID ip link set dev $CONTAINER_IFNAME address $MACADDR
+ip link set "$GUEST_IFNAME" netns "$NSPID"
+ip netns exec "$NSPID" ip link set "$GUEST_IFNAME" name "$CONTAINER_IFNAME"
+[ "$MACADDR" ] && ip netns exec "$NSPID" ip link set dev "$CONTAINER_IFNAME" address "$MACADDR"
 if [ "$IPADDR" = "dhcp" ]
 then
-  [ $DHCP_CLIENT = "udhcpc"  ] && ip netns exec $NSPID $DHCP_CLIENT -qi $CONTAINER_IFNAME -x hostname:$GUESTNAME
-  if [ $DHCP_CLIENT = "dhclient"  ]; then
+  [ "$DHCP_CLIENT" = "udhcpc"  ] && ip netns exec "$NSPID" "$DHCP_CLIENT" -qi "$CONTAINER_IFNAME" -x "hostname:$GUESTNAME"
+  if [ "$DHCP_CLIENT" = "dhclient"  ]; then
     # kill dhclient after get ip address to prevent device be used after container close
-    ip netns exec $NSPID $DHCP_CLIENT -pf "/var/run/dhclient.$NSPID.pid" $CONTAINER_IFNAME
+    ip netns exec "$NSPID" "$DHCP_CLIENT" -pf "/var/run/dhclient.$NSPID.pid" "$CONTAINER_IFNAME"
     kill "$(cat "/var/run/dhclient.$NSPID.pid")"
     rm "/var/run/dhclient.$NSPID.pid"
   fi
-  [ $DHCP_CLIENT = "dhcpcd"  ] && ip netns exec $NSPID $DHCP_CLIENT -q $CONTAINER_IFNAME -h $GUESTNAME
+  [ "$DHCP_CLIENT" = "dhcpcd"  ] && ip netns exec "$NSPID" "$DHCP_CLIENT" -q "$CONTAINER_IFNAME" -h "$GUESTNAME"
 else
-  ip netns exec $NSPID ip addr add $IPADDR dev $CONTAINER_IFNAME
+  ip netns exec "$NSPID" ip addr add "$IPADDR" dev "$CONTAINER_IFNAME"
   [ "$GATEWAY" ] && {
-    ip netns exec $NSPID ip route delete default >/dev/null 2>&1 && true
+    ip netns exec "$NSPID" ip route delete default >/dev/null 2>&1 && true
   }
-  ip netns exec $NSPID ip link set $CONTAINER_IFNAME up
+  ip netns exec "$NSPID" ip link set "$CONTAINER_IFNAME" up
   [ "$GATEWAY" ] && {
-    ip netns exec $NSPID ip route get $GATEWAY >/dev/null 2>&1 || \
-    ip netns exec $NSPID ip route add $GATEWAY/32 dev $CONTAINER_IFNAME
-    ip netns exec $NSPID ip route replace default via $GATEWAY
+    ip netns exec "$NSPID" ip route get "$GATEWAY" >/dev/null 2>&1 || \
+    ip netns exec "$NSPID" ip route add "$GATEWAY/32" dev "$CONTAINER_IFNAME"
+    ip netns exec "$NSPID" ip route replace default via "$GATEWAY"
   }
 fi
 
 # Give our ARP neighbors a nudge about the new interface
 if installed arping; then
-  IPADDR=$(echo $IPADDR | cut -d/ -f1)
-  ip netns exec $NSPID arping -c 1 -A -I $CONTAINER_IFNAME $IPADDR > /dev/null 2>&1 || true
+  IPADDR=$(echo "$IPADDR" | cut -d/ -f1)
+  ip netns exec "$NSPID" arping -c 1 -A -I "$CONTAINER_IFNAME" "$IPADDR" > /dev/null 2>&1 || true
 else
   echo "Warning: arping not found; interface may not be immediately reachable"
 fi
 
 # Remove NSPID to avoid `ip netns` catch it.
-[ -f /var/run/netns/$NSPID ] && rm -f /var/run/netns/$NSPID
-
-exit 0
+rm -f "/var/run/netns/$NSPID"
```

Comment:
`rm -f` is pretty much guaranteed to have a 0 exit status unless something really unexpected is going on. So I removed the `exit 0`, too. Hope that's OK.